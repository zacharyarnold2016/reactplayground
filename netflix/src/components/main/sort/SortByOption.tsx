import React from "react";
import { sortString } from "../../../helpers/sortHelperMethods";
import { useNavigate, useParams } from "react-router";
import generateUrl from "../../../helpers/generateUrlString";

const SortByOption = ({ option }: any) => {
  const searchState = useParams();
  const finalstring: string = sortString(option);
  const navigate = useNavigate();

  const sortAll = async () => {
    const { genre, searchQuery } = searchState;
    const newSearchState = { genre, searchQuery, sortBy: finalstring };
    const url = generateUrl(newSearchState);

    navigate(url);
  };

  return (
    <h5 style={{ padding: "1px" }} onClick={sortAll}>
      {option}
    </h5>
  );
};

export default SortByOption;
