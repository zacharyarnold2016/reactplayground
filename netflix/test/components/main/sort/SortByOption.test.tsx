// @ts-nocheck
import { expect } from "@jest/globals";
import renderer from "react-test-renderer";
import SortByOption from "../../../../src/components/main/sort/SortByOption";
import { useNavigate, useParams } from "react-router";
import generateUrl from "../../../../src/helpers/generateUrlString";

const spyFunction = jest.fn();
useNavigate = jest.fn(() => spyFunction);
generateUrl = jest.fn();

describe("SortByOption", () => {
  test("Snapshot", () => {
    const component = renderer.create(<SortByOption option="test" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("onClick", () => {
    const component = renderer.create(<SortByOption option="test" />);
    let tree = component.toJSON();
    tree.props.onClick();
    expect(generateUrl).toBeCalled();
  });
});
