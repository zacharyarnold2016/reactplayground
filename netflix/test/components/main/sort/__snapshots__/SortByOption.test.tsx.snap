// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SortByOption Snapshot 1`] = `
<h5
  onClick={[Function]}
  style={
    Object {
      "padding": "1px",
    }
  }
>
  test
</h5>
`;
